proxy/http/HttpSM.cc

```diff
@@ -974,19 +973,34 @@ HttpSM::state_watch_for_client_abort(int event, void *data)
    */
   case VC_EVENT_EOS: {
     // We got an early EOS. If the tunnal has cache writer, don't kill it for background fill.
-    NetVConnection *netvc = ua_txn->get_netvc();
-    if (ua_txn->allow_half_open() || tunnel.has_consumer_besides_client()) {
-      if (netvc) {
-        netvc->do_io_shutdown(IO_SHUTDOWN_READ);
+    if (!terminate_sm) { // Not done already
+      NetVConnection *netvc = ua_txn->get_netvc();
+      /** TODO: we want to put tunnel.has_consumer_besides_client back in here.
+       *
+       * Currently, our HTTP/2 to origin branch does not work with background
+       * fill.  Trying to keep the tunnel around to fill the cache after the
+       * client aborts currently results in a connection leak. This has to be
+       * fixed, but for now we're avoiding this relatively new feature (Spring
+       * 2021) to stop the leak.
+       *
+       * Uncomment this code to reinstate background fill once it is fixed.
```

Comment:
It's been over a year since this TODO comment. Any progress?

Reply:
@duke8253 is currently working on fixing this. Our hope/intention is to fix this as an additional patch on this PR before merging.

Reply:
By the way, CI is currently failing the autest for background_fill precisely because of this issue. So, to be clear, I don't intend to move forward with this PR until we get this feature working.

Reply:
@duke8253 provided a fix for background fill which I have now incorporated into this PR.